controllers/operator/routes.go

```diff
@@ -247,6 +247,10 @@ func (r *AuthenticationReconciler) handleRoutes(ctx context.Context, instance *o
 	return
 }
 
+func shouldHaveRoutes(instance *operatorv1alpha1.Authentication) bool {
+	return !instance.Spec.Config.ZenFrontDoor
```

Comment:
If the flag doesn't exist, it will default to its zero value, `false`, which is the default for the field. This seems fine to me, but let me know if you disagree.